### ISSUE TITLE
Feature: insights / lists - Add breadcrumbs

### ISSUE THREAD

**Issue by jpmcb** [open-sauced/app#1822]
### Type of feature

🍕 Feature

### Current behavior

When in a list, and this goes for insight pages as well, I keep finding myself wanting to go back to view other lists.

But since there's no breadcrumb back to the lists overview, I end up clicking --> Insights tab --> Manually entering "lists" to get to lists in the url bar --> view my lists

A breadcrumb would be super ideal for getting back to the list / insights overview page

### Suggested solution

<img width="1360" alt="Screenshot 2023-10-06 at 8 55 50 AM" src="https://github.com/open-sauced/app/assets/23109390/f23a6fa4-8aa8-4a09-9db1-392cd34b3470">

Rough sketch of where a breadcrumb might be

cc @isabensusan - thoughts?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by isabensusan** [comment]
I've tried a couple designs and it kind of breaks the layout whatever we do. I'll put a pin on it for now until I find a better way to tackle this

**Comment by isabensusan** [comment]
@jpmcb @nickytonline sorry for the delay on this, but what do you think of a breadcrumb component instead of a "back" button? 
Because we intend to make the platform more complex, I think having the breadcrumbs is going to scale better as we add complexity to the navigation.

<img width="1837" alt="image" src="https://github.com/open-sauced/app/assets/27322879/3f7b08cf-46ea-40e2-87ff-8538fbd3e63c">

close up:
<img width="698" alt="image" src="https://github.com/open-sauced/app/assets/27322879/2d58e1a4-49d7-4a06-a9d2-854cbc6cd327">

**Comment by nickytonline** [comment]
Yes please to breadcrumbs over a back button in the site on pages. 😎

**Comment by bdougie** [comment]
This would 100% be useful in the Workspaces and Insights today.
